src/Native/System.Native/pal_io.cpp

```diff
@@ -443,3 +464,93 @@ int32_t MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     return mkstemps(pathTemplate, suffixLength);
 }
+
+static
+void ConvertPalPollFDToNative(const PollFD& pal, pollfd& native)
+{
+    native.fd = pal.FileDescriptor;
+    native.events = pal.EventsToPollFor;
+    native.revents = pal.ReceivedEvents;
+}
+
+static
+void ConvertNativePollfdToPal(const pollfd& native, PollFD& pal)
+{
+    pal.FileDescriptor = native.fd;
+    pal.EventsToPollFor = native.events;
+    pal.ReceivedEvents = native.revents;
+}
+
+extern "C"
+int32_t Poll(PollFD* pollData, uint32_t numberOfPollFds, int32_t timeout)
+{
+    assert(numberOfPollFds <= 2);
+
+    // Convert our standardized pollfd to the native one
+    pollfd fds[2];
+    for (int i = 0; i < numberOfPollFds; i++)
+    {
+        ConvertPalPollFDToNative(pollData[i], fds[i]);
+    }
+
+    // Call poll with the native pollfd and, if necessary, convert the results back to our standardized pollfd
+    int32_t result = poll(fds, numberOfPollFds, timeout);
+    if (result > 0) // We only have result data if the result is positive
+    {
+        for (int i = 0; i < numberOfPollFds; i++)
+        {
+            ConvertNativePollfdToPal(fds[i], pollData[i]);
+        }
+    }
+
+    return result;
+}
+
+extern "C"
+int32_t PosixFAdvise(int32_t fd, int64_t offset, int64_t length, FileAdvice advice)
+{
+#if HAVE_POSIX_ADVISE
+    return posix_fadvise(fd, offset, length, (int)advice);
+#else
+    // Not supported on this platform; however, we don't want to #error here since
+    // currently, this isn't used on platforms where it isn't supported.
+    return ENOSYS;
```

Comment:
I've been using ENOTSUP for this. Would be good to stay consistent. Is ENOSYS better?